protos/encodings.proto

```diff
@@ -161,6 +161,14 @@ message Flat {
   uint64 bits_per_value = 1;
   // the buffer of values
   Buffer buffer = 2;
+
+  // additional metadata that should be present if bitpacking is used
+  optional BitpackMeta bitpack_meta = 4;
+}
```

Comment:
Minor nit: I think of bitpacking less as an extension of `Flat` and more as it's own encoding that has another array encoding inside of it (like `fixed_size_list`).  I don't know of any concrete reason that's better but I like thinking of these as small composable pieces rather than one piece with lots of options.

Reply:
good call, made this change